packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -345,7 +345,7 @@ class ThemeData with Diagnosticable {
     textSelectionHandleColor ??= isDark ? Colors.tealAccent[400]! : primarySwatch[300]!;
     backgroundColor ??= isDark ? Colors.grey[700]! : primarySwatch[200]!;
     dialogBackgroundColor ??= isDark ? Colors.grey[800]! : Colors.white;
-    indicatorColor ??= accentColor == primaryColor ? Colors.white : accentColor;
+    indicatorColor ??= accentColor;
```

Comment:
I saw that you explained it in the PR description, but I don't quite understand what you mean since it's still possible for both accentColor and pimaryColor to be a `Color`, which would cause the theme default indicator color to be `Colors.white`. Any apps expecting this behavior would break.

Reply:
I understand what you mean.
Imagine such a situation, if the `pimaryColor` is `MaterialColor(primary value: Color(0xff2196f3))` or `Color(0xff2196f3))` and the `accentColor` is `Color(0xff2196f3))`, Visually they are the same color but have different indicator colors. This behavior is very strange and incomprehensible. What I think is that if we haven't broken something now, we can fix it to avoid this weird behavior in the future.

Of course, this issue has nothing to do with this PR, I will open another topic to discuss this, thank you for your suggestions.

Reply:
Track at #69162